test/integration/controller/job/job_controller_test.go

```diff
@@ -127,6 +131,26 @@ var _ = ginkgo.Describe("Job controller", func() {
 			err := k8sClient.Get(ctx, lookupKey, createdWorkload)
 			return err == nil
 		}, consistentDuration, interval).Should(gomega.BeTrue())
+		events := &v1.EventList{}
+		gomega.Eventually(func() bool {
```

Comment:
This doesn't seem correct.

Is empty list a failure? Even if it's not, just waiting for List to succeed is not a guarantee that the list will have the event we are looking for. Specially on the 2nd check (line 171).

The checking of the list should be inside the `gomega.Eventually` call.

An alternate implementation is using watchers, similar to https://github.com/kubernetes-sigs/kueue/blob/main/pkg/scheduler/scheduler_test.go#L650-L667

Reply:
Sorry. I am not sure if I understand correctly. I check the event in L141 to make sure it’s what we want.

Reply:
Yes, but if it's not satisfied, you should reload list again. There is no guarantee that by the time you call List, the event that you want is already there.

Reply:
right, you need to move the logic in the Expect function below here.

Reply:
Thanks. Got it. Fixed.

Reply:
Can you refactor into a function that can check any type of event?

Maybe you can have it pkg/util/testing.